app/models/user.rb

```diff
@@ -9,5 +9,12 @@ class User < ApplicationRecord
   has_secure_password
 
   has_many :friendships
-  has_many :friendships, through: :frienships
+  has_many :friendships, through: :friendships
+
+  def self.check_for_username?(username)
+    users = User.where(username: username)
+    return false if users.size == 0
+
+    true
+  end
```

Comment:
can we instead use User.where(username: username).nil?

Reply:
Good catch, I'll change that

Reply:
I changed it to use .empty? instead of users.size == 0, but kept the conditional the same to have the method return true when the username is in our database.